docs/upgrade-to-6.md

```diff
@@ -0,0 +1,649 @@
+  "dependencies": {
+-   "@gasket/intl": "^5.0.0",
++   "@gasket/react-intl": "^6.0.0"
+    "@gasket/plugin-intl": "^6.0.0"
+    ...
+  }
+}
+```
+
+### Simplified locale paths
+
+We now have a much simpler approach for specifying the paths to load locale
+files from. These were previously referred to as "identifiers" in the old
+documentation. Now, we call them the "localesPath", and they are just a string.
+
+```diff
+- <LocaleRequired identifier={{ namespace: 'namespace' }}>
++ <LocaleRequired localesPath='/locales/namespace'>
+  ...
+</LocaleRequired>
+```
+
+For a more detailed explanation, refer to the [localesPath docs][localesPath].
+
+### Next.js Initial Props
+
+To allow for a static generation of pages, the `withLocaleRequired` HOC no
+longer automatically adds `getInitialProps` to the component. It will still
+fetch the locale file, but this will happen in the browser only now.
+
+However, if you still want to server-render pages with locale files preloaded,
+you can re-enable `getInitialProps` with the HOC `initialProps` option.
+
+```diff
+import { withLocaleRequired } from '@gasket/react-intl';
+import { FormattedMessage } from 'react-intl';
+
+const Component = props => <h1><FormattedMessage id='welcome'/></h1>
+
+- export default withLocaleRequired('/locales')(Component);
++ export default withLocaleRequired('/locales', { initialProps: true })(Component);
+```
+
+## Static Progressive Web App Changes
+
+### Update Lifecycle Signatures with Context Object
+
+To better support static site generation, we have updated the
+[@gasket/plugin-service-worker] and [@gasket/plugin-workbox] lifecycles'
+signatures to pass in a context object instead of the usual `req, res`.
+
+```diff
+// SERVICE WORKER USAGE EXAMPLE
+
+- module.exports = async function composeServiceWorker(gasket, content, req) {
++ module.exports = async function composeServiceWorker(gasket, content, { req }) {
+```
+
+```diff
+// WORKBOX USAGE EXAMPLE
+
+- module.exports = function workbox(gasket, config, req) {
++ module.exports = function workbox(gasket, config, { req }) {
+```
+
+The purpose for using this context object is to allow these lifecycle methods to
+be run at build time without a request object, or at run time, when the request
+object is present.
+
+_Impacted Plugins/Packages: `@gasket/plugin-service-worker`,
+`@gasket/plugin-workbox`, `@gasket/plugin-manifest`_
+
+## Webpack 5
+
+Webpack 5 support is now available! We have tuned all of our plugins and
+packages to fully support Webpack 5.
+
+> Upgrading to Webpack 5 is not a required prerequisite for upgrading Gasket.
+
+Update `webpack` to v5.
+
+```diff
+"devDependencies": {
+-   "webpack": "^4.44.1",
++   "webpack": "^5.9.0"
+}
+```
+
+### Removed Config Defaults
+
+We have removed the generated Webpack config defaults. The `node` config options
+that were previously prescribed, have changed. You can find more info about
+configuring Node.js options with Webpack 5 on the [webpack website].
```

Comment:
I have a suggestion branch for reworking this section. See https://github.com/godaddy/gasket/compare/v6-upgrade-guide...v6-upgrade-guide-webpack

You can `cherry-pick` the commit over if you like it.